### DISCUSSION TITLE
Use ADSideVectorDiffusiveFluxIntegral

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#26886] [Q&A General] [answered]
I am attempting to use the following predefined postprocessor, ADSideVectorDiffusivityFluxIntegral, I am getting the following error shown bellow. What do I need to do to use this postprocessor?

*** ERROR ***
The following error occurred in the object "MOOSE Problem", of type "FEProblem".

No standard variable named velocity found. Did you specify a vector variable when you meant to specify a standard variable?

**Comment by GiudGiud** [comment]
Hello
Hello

Could you please paste here your Variables and AuxVariables block, as well as the relevant postprocessor 

Guillaume

**Reply by GiudGiud** [reply, marked as answer]
I think what's happening is that you are trying to use this with a vector variable. But this postprocessor uses:
- a regular variable, forms the gradient
- an anisotropic diffusivity, which is the `VectorDiffusive` part of the name

and does the product with the normal to compute the diffusive flux
So if you want to use a velocity vector variable, you need to create a new postprocessor that uses a vector variable and does this operation component by component i imagine. 

Guillaume

**Reply by TLWise** [reply]
@GiudGiud , So it seems that I can define an aux variable vel_x from the x component of velocity. I could then use ADSideVectorDiffusivityFluxIntegral or similar to get the solution of the intergral of the following mu*(d(vel_x)/dx*n_x+d(vel_x)/dy*n_y)?

**Reply by GiudGiud** [reply]
Yes. (d(vel_x)/dxn_x+d(vel_x)/dyn_y) = grad_vel_x * normal

**Reply by GiudGiud** [reply]
but if mu is not a vector, you dont need the `ADSideVECTORDiffusivityFluxIntegral`. just use the regular one

**Reply by TLWise** [reply]
Is there a predefined way to get  mu*(d(vel_x)/dx*n_x+d(vel_y)/dy*n_x)?

**Reply by GiudGiud** [reply]
not that I know

**Reply by TLWise** [reply]
@GiudGiud Thanks for the continued assistance. Let me know if you think of something novel to solve this problem.

**Reply by TLWise** [reply]
@GiudGiud Where in the MOOSE code is _normals defined?

**Reply by GiudGiud** [reply]
In the base class, if you look at line 40 of SideUserObject
```
    _normals(_assembly.normals()),
```
